Description: Compile the bootstrap compiler with a limited (empirically
determined) number of jobs on Arm64. Ninja defaults to NCPU jobs (NCPU being
the number of CPUs the system has). This was observed to exhaust the memory
of Arm64 Launchpad builders, triggering the OOM killer.
Author: Nathan Pratta Teodosio <nathan.teodosio@canonical.com>
Forwarded: not-needed

--- a/tools/clang/scripts/build.py
+++ b/tools/clang/scripts/build.py
@@ -746,6 +746,10 @@
         # Ignore args.disable_asserts for the bootstrap compiler.
         '-DLLVM_ENABLE_ASSERTIONS=ON',
     ]
+    if sys.platform.startswith('linux') and platform.machine() == 'aarch64':
+      bootstrap_args.extend([
+          '-DLLVM_PARALLEL_COMPILE_JOBS=2',
+          ])
     if sys.platform == 'darwin':
       # On macOS, the bootstrap toolchain needs to have compiler-rt because
       # dsymutil's link needs libclang_rt.osx.a. Only the x86_64 osx
